src/java/test/org/apache/zookeeper/server/ZooKeeperServerMainTest.java

```diff
@@ -133,6 +138,62 @@ public void shutdown() {
     }
 
     /**
+     * Test case for https://issues.apache.org/jira/browse/ZOOKEEPER-2247.
+     * Test to verify that even after non recoverable error (error while
+     * writing transaction log) on ZooKeeper service will be available
+     */
+    @Test(timeout = 30000)
+    public void testNonRecoverableError() throws Exception {
```

Comment:
ZooKeeperServerMainTest.java test class to verify standalone cases. The newly added test is to verify the behavior in standalone server. So this test case is needed for us.

NonRecoverableErrorTest.java verifies quorum behavior. I will add class level javadoc in this class to make it more clear, is that ok?